src/ledger/Reader.zig

```diff
@@ -2681,3 +2691,34 @@ test getConfirmedSignaturesForAddress {
     try std.testing.expect(sig_infos.infos.items.len > 0);
     try std.testing.expect(sig_infos.found_before);
 }
+
+test "isDuplicateSlot" {
+    const allocator = std.testing.allocator;
+    var registry = sig.prometheus.Registry(.{}).init(allocator);
+    defer registry.deinit();
+
+    var state = try sig.ledger.tests.initTestLedger(allocator, @src(), .FOR_TESTS);
+    defer state.deinit();
+
+    const reader = state.reader();
+    const result_writer = state.resultWriter();
+
+    // Test case: Slot with no duplicate proof returns false
+    {
+        const slot: Slot = 42;
+        const is_duplicate = try reader.isDuplicateSlot(slot);
+        try std.testing.expectEqual(false, is_duplicate);
```

Comment:
Can give a more informative error message that reflects the fact that we "expect" this to be false, and not an error. Not always applicable, but nice to do when possible.
```suggestion
        try std.testing.expectEqual(false, reader.isDuplicateSlot(slot));
```

Reply:
[eb9ca4a](https://github.com/Syndica/sig/pull/1027/commits/eb9ca4a660a2e0b3a3bbd007ea705a26d86d3b64)